src/Microsoft.Framework.PackageManager/Building/BuildManager.cs

```diff
@@ -34,9 +41,63 @@ public BuildManager(IServiceProvider hostServices, BuildOptions buildOptions)
 
         public bool Build()
         {
+            var projectFilesFinder = new Matcher();
+
+            // Resolve all the project names
+            var projectFilesToBuild = new List<string>();
+            foreach(var pattern in _buildOptions.ProjectPatterns)
+            {
+                if (pattern.Contains("*"))
+                {
+                    // Requires globbing
+                    projectFilesFinder.AddInclude(NormalizeGlobbingPattern(pattern));
+                }
+                else
+                {
+                    projectFilesToBuild.Add(pattern);
+                }
+            }
+
+            var rootDirectory = Directory.GetCurrentDirectory();
+            var patternSearchFolder = new DirectoryInfoWrapper(new DirectoryInfo(rootDirectory));
+            var globbingProjects = projectFilesFinder.Execute(patternSearchFolder).Files.Select(file =>
+                Path.Combine(rootDirectory, file));
+            projectFilesToBuild.AddRange(globbingProjects);
+
+            var sw = Stopwatch.StartNew();
+
+            _cacheContextAccessor = new CacheContextAccessor();
+            _cache = new Cache(_cacheContextAccessor);
+
+            var globalSucess = true;
+            foreach (var project in projectFilesToBuild)
+            {
+                var buildSuccess = BuildInternal(project);
+                globalSucess &= buildSuccess;
+            }
+
+            _buildOptions.Reports.Information.WriteLine($"Total build time elapsed: { sw.Elapsed }");
+            _buildOptions.Reports.Information.WriteLine($"Total projects built: { projectFilesToBuild.Count }");
+
+            return globalSucess;
+        }
+
+        private static string NormalizeGlobbingPattern(string pattern)
+        {
+            // TODO: use a constant for project.json
+            if (!string.Equals(Path.GetFileName(pattern), "project.json", StringComparison.OrdinalIgnoreCase))
+            {
+                return pattern + "/project.json";
+            }
+
+            return pattern;
+        }
+
+        private bool BuildInternal(string projectPath)
+        {
             var projectDiagnostics = new List<ICompilationMessage>();
-            Runtime.Project project;
-            if (!Runtime.Project.TryGetProject(_buildOptions.ProjectDir, out project, projectDiagnostics))
+
+            if (!Runtime.Project.TryGetProject(projectPath, out _currentProject, projectDiagnostics))
```

Comment:
Could this not be a local variable anymore? It only seems to be used in this method.